tests/functional/test_login_command.py

```diff
@@ -83,5 +83,34 @@ def test_login_gcs_different_identity(
     monkeypatch.setattr(
         "globus_cli.commands.logout.internal_native_client", lambda: mock_auth_client
     )
-    result = run_line("globus logout --yes")
+    run_line("globus logout --yes")
     assert test_token_storage.read_config(_STORE_CONFIG_USERINFO) is None
+
+
+def test_login_with_flow(monkeypatch, run_line):
+    """Verify that flow ID's are added as resource servers with correct scopes."""
+
+    uuid1 = str(uuid.uuid4())
+    scope_name = f'flow_{uuid1.replace("-", "_")}_user'
```

Comment:
Do you want to replace this scope-string building with the SDK method as well? I can see arguments either way, but my initial take is that we want to avoid having `.replace("-", "_")` anywhere in this repo. 😸